game-core/src/main/java/games/strategy/engine/data/DefaultAttachment.java

```diff
@@ -67,6 +68,16 @@ protected static int getInt(final String value) {
     }
   }
 
+  /** Throws an error if format is invalid. */
+  protected static BigDecimal getBigDecimal(final String value) {
```

Comment:
This tool seems to be in an odd place, and it's only used by one child class. What do you think about making this a private static utility function or moving it to a utils class? `java-extras` would be a good hope, perhaps something like `BigDecimalFactory`

Reply:
Actually many of the methods of this class are helpers for reading XML strings and converting them to the appropriate types.